### PR TITLE
chore: fix cyclic dependency in repo caused by generate-lib script changes

### PR DESCRIPTION
**Pull request by bradzacher**
## Overview

In #7535 I made a change to the script so that it used our `utils` package to type an `ESLint` call.
This actually introduced a nx cyclic dependency crash into the repo - but it looks like this crash was never seen due to caching or something I guess?

However on a fresh machine if you ran `yarn`, then it'll cause yarn to attempt to do a fresh build after an install and then nx would fail with
```
NX Could not execute command because the task graph has a circular dependency rule-schema-to-typescript-types:build --> type-utils:build --> utils:build --> scope-manager:build --> utils:build
```

cc @JamesHenry - this behaviour is super weird!

This PR moves the script to the `repo-tools` folder so that it's an isolated leaf, fixing the cyclic dependency.

### CODE REVIEW

**Review by JoshuaKGoldberg — DISMISSED** [review]
Heh, I just started gettin this too - thanks for the fix. I'll defer to @JamesHenry as the Nx wiz.

(assuming the lockfile is fixed ofc)

**Comment by JoshuaKGoldberg** [comment]
or actually, @bradzacher this feels like a good thing to get in asap - I'm happy to just get it in sooner

**Comment by bradzacher** [comment]
Yeah we can land it sooner then cycle back with James to investigate why this wasn't caught sooner by nx.

I'm not going to be able to get to my laptop for around 2 hours - if you wanted it asap feel free to pull and fix the lock file for me :)

**Comment by JoshuaKGoldberg** [comment]
Well I'm stumped...

```plaintext
Internal Error: @typescript-eslint/typescript-eslint@workspace:.: This package doesn't seem to be present in your lockfile; run "yarn install" to update the lockfile
    at V0.getCandidates (/Users/josh/.cache/node/corepack/yarn/3.6.3/yarn.js:435:5145)
    at kf.getCandidates (/Users/josh/.cache/node/corepack/yarn/3.6.3/yarn.js:391:1264)
    at /Users/josh/.cache/node/corepack/yarn/3.6.3/yarn.js:439:8033
    at df (/Users/josh/.cache/node/corepack/yarn/3.6.3/yarn.js:390:11070)
    at ge (/Users/josh/.cache/node/corepack/yarn/3.6.3/yarn.js:439:8013)
    at async Promise.allSettled (index 0)
    at async io (/Users/josh/.cache/node/corepack/yarn/3.6.3/yarn.js:390:10398)
    at async /Users/josh/.cache/node/corepack/yarn/3.6.3/yarn.js:439:8545
    at async ti.startProgressPromise (/Users/josh/.cache/node/corepack/yarn/3.6.3/yarn.js:390:47579)
    at async je.resolveEverything (/Users/josh/.cache/node/corepack/yarn/3.6.3/yarn.js:439:6573)
```

**Comment by bradzacher** [comment]
Nuke a bitch and try again? That's my standard workflow for fixing yarn problems heh

**Comment by JoshuaKGoldberg** [comment]
No juice :(

**Review by armano2 — APPROVED** [review]

**Comment by bradzacher** [comment]
okay so I now have a fresh and clean `yarn` working.
I had to migrate our patches off of `patch-package` and onto `yarn patch` (which is now built into yarn2).

File: yarn.lock [hunk 1]
```diff
@@ -6523,6 +6522,18 @@ __metadata:
...
+  resolution: "ajv@patch:ajv@npm%3A6.12.6#./.yarn/patches/ajv-npm-6.12.6-4b5105e2b2.patch::version=6.12.6&hash=3d7604&locator=%40typescript-eslint%2Ftypescript-eslint%40workspace%3A."
+  dependencies:
+    fast-deep-equal: ^3.1.1
+    fast-json-stable-stringify: ^2.0.0
+    json-schema-traverse: ^0.4.1
+    uri-js: ^4.2.2
+  checksum: 89f87655eef3d74b782ccd96fbb957987a905189844f293754371f28b7795f21153786c39707002bee2eb24c23c8a6efd8a9a89365afcca7d6e50df573bf5047
+  languageName: node
+  linkType: hard
+
```
Comment (bradzacher) [inline yarn.lock:6536]:
```suggestion
```

**Comment by bradzacher** [comment]
GH statuses didn't update - but there was a build flake.
Retried the deploy and it worked fine
https://app.netlify.com/sites/typescript-eslint/deploys/64f91a3b3e09390d067e94a7
(the netlify comment updated though...)

**Pull request merged**